src/main/java/seedu/todo/ui/TaskCard.java

```diff
@@ -107,26 +89,52 @@ public String getFxmlPath() {
     }
     
     /**
-     * Displays the tags with tag labels sorted lexicographically.
+     * Displays the tags in lexicographical order, ignoring case.
      */
     private void displayTags(){
-        //Obtain and verify that there are at most 5 tags.
+        Label[] tagLabels = {tagLabel0, tagLabel1, tagLabel2, tagLabel3, tagLabel4};
+        
         int numberOfTags = task.getTags().size();
         assert (numberOfTags <= 5);
         
-        //Extract tag names, and sort the tags lexicographically.
-        Stream<String> tagsStream = task.getTags().stream()
-                .map(tag -> tag.tagName);
-        tagsStream = tagsStream.sorted();
-        String[] tags = (String[]) tagsStream.toArray();
+        LinkedList<Tag> tagList = new LinkedList<>(task.getTags());
+        tagList.sort(new Comparator<Tag>(){
+            @Override
+            public int compare(Tag o1, Tag o2) {
+                return o1.toString().compareToIgnoreCase(o2.toString());
+            }
+        });
         
-        //Label or hide the tagLabels
-        for (int i = 0; i < 5; i++) {
-            if (i < numberOfTags) {
-                tagLabels[i].setText(tags[i]);
+        for (Label label : tagLabels) {
+            if (tagList.isEmpty()) {
+                FxViewUtil.setCollapsed(label, true);
             } else {
-                tagLabels[i].setVisible(false);
+                label.setText(tagList.poll().tagName);
             }
         }
     }
+    
+    private void setStyle() {
+        ObservableList<String> stylesheets = taskCard.getStylesheets();
+        stylesheets.add(STYLE_BASE);
+        //stylesheets.add(STYLE_COLLAPSED); Disabled until implemented
+        
+        if (task.isCompleted()) {
```

Comment:
This seems like it would have very bad performance, as you're reloading CSS on a per element basis on the fly. Have you considered putting all of the styles in one stylesheet, load that globally and adding/removing classes instead for switching between state styles? Also, would this render correctly if I push back the due date on a task?